torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -107,6 +107,10 @@ Value* tryConvertToType(
         DeviceObjType::get()->isSubtypeOf(concrete_type)) {
       return graph.insert(aten::device, {value}, {}, loc);
     }
+    if (concrete_type == FloatType::get() &&
```

Comment:
Looking at this PR now because of the bug James commented about. But this line is suspect as well because it broadens semantics. Why do we allow implicit conversions from Number -> float but not from Number -> int? Both of those are lossy conversions.

@driazati - changes to implicit conversions should always have high scrutiny. We need to make sure there is a discussion before a change is landed.